UnityProject/Assets/Scripts/Systems/Explosions/NodeTypes/ExplosionNode.cs

```diff
@@ -68,57 +65,92 @@ public virtual void Process()
 		}
 
 		//method that, surprise, does damage to stuff on node's tile. override for custom behaviour. must return EnergyExpended value
-		public virtual float DoDamage(Matrix matrix, float damageDealt, Vector3Int v3int)
+		public virtual float DoDamageToTiles(Matrix matrix, float damageDealt, Vector3Int v3int, MetaTileMap metaTileMap)
 		{
-			var metaTileMap = matrix.MetaTileMap;
 			float energyExpended = metaTileMap.ApplyDamage(v3int, damageDealt,
 			MatrixManager.LocalToWorldInt(v3int, matrix.MatrixInfo), AttackType.Bomb);
 
 			DamageLayers(damageDealt, v3int);
+			ChangeNodeTemp(matrix, damageDealt, v3int);
+
+			return energyExpended;
+		}
 
+		public virtual async UniTask DoDamageToObjects(Matrix matrix, float damageDealt, Vector3Int v3int)
+		{
 			foreach (var integrity in matrix.Get<Integrity>(v3int, true))
 			{
-				//Throw items and Objects
-				integrity.GetComponent<UniversalObjectPhysics>()?.NewtonianNewtonPush(AngleAndIntensity.Rotate90(), AngleAndIntensity.magnitude * 0.1f , 1, 3,
-					BodyPartType.Chest, integrity.gameObject, 15);
+				// Incase of multiple explosions occuring at once (i.e: multiple Gibtonites)
+				// trycatch this to avoid trying to destroy stuff that is already destroyed by other damage sources.
+				try
+				{
+					integrity.Physics.NewtonianNewtonPush(AngleAndIntensity.Rotate90(), AngleAndIntensity.magnitude * 0.1f , 1, 3,
+						BodyPartType.Chest, integrity.gameObject, 15);
+
+					if (IgnoreAttributes != null)
+					{
+						await UniTask.WaitForEndOfFrame();
+						if (integrity.TryGetComponent<ItemAttributesV2>(out var traits) &&
+						    traits.HasAnyTraitZeroAlloc(IgnoreAttributes)) continue;
+					}
 
-				if (integrity.TryGetComponent<ItemAttributesV2>(out var traits))
+					// And do damage to objects
+					integrity.ApplyDamage(damageDealt, AttackType.Bomb, DamageType.Brute);
+				}
+				catch (Exception e)
 				{
-					if (IgnoreAttributes != null && traits.HasAnyTrait(IgnoreAttributes)) continue;
+					Loggy.Error($"An error occured while trying to damage an object. Maybe it's no longer avaliable?\n {e}");
 				}
-
-				//And do damage to objects
-				integrity.ApplyDamage(damageDealt, AttackType.Bomb, DamageType.Brute);
 			}
 
+			// Damage mobs
 			foreach (var player in matrix.Get<LivingHealthMasterBase>(v3int, ObjectType.Player, true))
 			{
-				player.GetComponent<UniversalObjectPhysics>()?.NewtonianPush(AngleAndIntensity.Rotate90(), 7, 1, 3,
-                					BodyPartType.Chest, player.gameObject, 15);
-				// do damage
-				player.ApplyDamageAll(null, damageDealt, AttackType.Bomb, DamageType.Brute, default, TraumaticDamageTypes.NONE, 75);
+				//Player damage is relatively fine performance wise, and doesn't generate any GC.
+				//However, it is still a bit too complex; and its complexity shows when several mobs are recieving damage at once.
+				//Processing one at a time each frame should give the game plenty of breathing room.
+				await UniTask.WaitForEndOfFrame();
```

Comment:
I presume this does nothing to affect the actual stepping program right|? 
so it wouldn't slow down the explosion if there's a bunch of mobs in one spot?

Reply:
It doesn't stop the execution because it's asynchronous. All processes will continue even when one is taking some time oofing some players.